plugins/graph_algorithm/src/algorithms/components.cpp

```diff
@@ -0,0 +1,73 @@
+#include "graph_algorithm/algorithms/components.h"
+
+#include "graph_algorithm/netlist_graph.h"
+
+#include <map>
+
+namespace hal
+{
+    namespace graph_algorithm
+    {
+        Result<std::vector<std::vector<u32>>> get_connected_components(NetlistGraph* graph, bool strong, u32 min_size)
+        {
+            if (graph == nullptr)
+            {
+                return ERR("graph is a nullptr");
+            }
+
+            igraph_vector_int_t membership, csize;
+            igraph_integer_t number_of_clusters;
+            auto err = igraph_vector_int_init(&membership, 0);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                return ERR(igraph_strerror(err));
+            }
```

Comment:
```suggestion
            if (err != IGRAPH_SUCCESS)
            {
                return ERR(igraph_strerror(err));
            }
```

Do not destroy igraph data structures whose initiailzation failed. They have never been created, so they don't need to be destroyed. We do not ensure that trying to do so won't cause problems (such as a crash).

Reply:
fixed (as are all the other things below as well). Thanks for taking the time and reviewing the code :D Also, thank you for maintaining igraph!